lxd/storage_pools_utils.go

```diff
@@ -117,12 +117,6 @@ func storagePoolCreateLocal(state *state.State, poolID int64, req api.StoragePoo
 
 	revert.Add(func() { _ = pool.Delete(clientType, nil) })
 
-	// Mount the pool.
```

Comment:
I understand you can remove the call to `ensureInitialDatasets` from the zfs drivers `Create()` as this is already called when running `Mount()`. 

But this change seems rather powerful as if affects all of the drivers. Does the zfs driver even have to mount the pool if it's anyway done here after calling the drivers `Create()`?

Reply:
Yes I agree this change might be too powerful, this is why I am leaving this as a draft. I just pushed some changes leaving `storagePoolCreate` unchanged and removing the `Mount()` call from `*lxdBackend.Create` and `ensureInitialDatasets()` from the zfs drivers `Create()`. If you agree this makes sense I will open the PR for review.

Reply:
Also I thought it made sense to discard the cherry picked commit since now my changes are following a very different logic, please let me know if that was a mistake

Reply:
This breaks a few of the storage drivers if you check the pipeline.

I would not try to fix this in the pools backend, instead as the problem is only occuring for the zfs driver, putting in a fix that only affects the zfs driver seems more reasonable to me.

Can you maybe make `ensureInitialDatasets()` idempotent and skip its actions if already done?